src/qth_locator.py

```diff
@@ -64,24 +54,28 @@ def square_to_location(qth_locator):
     lon += 5.0 / 60 * lon_sub_sq
     lat += 2.5 / 60 * lat_sub_sq
 
-    lon += 0.5 / 60 * lon_ext_sq
-    lat += 0.25 / 60 * lat_ext_sq
+    lat = round(lat, 6)
+    lon = round(lon, 6)
 
     return lat, lon
 
 
-def location_to_square(lat, lon):
+def location_to_square(lat: float, lon: float) -> None | str:
     """
     Converts latitude and longitude in decimal format to QTH locator.
     Gets latitude and longitude as floats.
     Returns QTH locator as string.
     """
 
     # Validate input
-    assert isinstance(lat, (int, float))
-    assert isinstance(lon, (int, float))
-    assert -90.0 <= lat <= 90.0
-    assert -180.0 <= lon <= 180.0
+    try:
+        assert isinstance(lat, float)
+        assert isinstance(lon, float)
+        assert -90.0 <= lat <= 90.0
+        assert -180.0 <= lon <= 180.0
+    except AssertionError as e:
+        # print(f"Latitude or Longitude is wrong\n{e}")
+        return None

```

Comment:
I would let the code explode on validation error instead. It makes sense to alert on input errors rather than return None in my opinion. There is also lesser to no point in throwing an exception and catching it one line below, because then you can rather use `if/else` and some guard statement.

The exception can help you to build API error handler too. So if one function bombs and the API can propagate that error towards response to the user as BadRequest.

Reply:
Good catch 👍 
Fixed in: https://github.com/Pytlicek/hamradio_functions/pull/8/commits/c50dc3120a26683bd073ad94f904699bee7e5120